services/dynamic_addressbook_add_node.proto

```diff
@@ -0,0 +1,79 @@
+syntax = "proto3";
+
+package proto;
+
+/*-
+ * ‌
+ * Hedera Network Services Protobuf
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+option java_package = "com.hederahashgraph.api.proto.java";
+// <<<pbj.java_package = "com.hedera.hapi.node.consensus">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+import "basic_types.proto";
+
+/**
+ * See [DynamicAddressBookService.addNode()](#proto.DynamicAddressBookService)
+ */
+message DynamicAddressBookAddNodeTransactionBody {
+    /**
+     * The unique entity id of the Node.
+     */
+    int64 nodeID = 1;
+    /**
+     * node account id
+     */
+    AccountID accountID = 2;
+    /**
+     * description of the node
+     */
+    string description = 3;
+    /**
+     * inbound ip address or FQDN
+     */
+    string inboundIp = 4;
+    /**
+     * inbound port
+     */
+    int32 inboundPort = 5;
+    /**
+     * outbound ip address or FQDN
+     */
+    string outboundIp = 6;
+    /**
+     * outbound port
+     */
+    int32 outboundPort = 7;
+    /**
+     * grpc ip address or FQDN
+     */
+    string grpcIp = 8;
+    /**
+     * grpc port
+     */
+    int32 grpcPort = 9;
```

Comment:
This flattened structure doesn't allow room for having arbitrary number of proxy endpoints. Also, it doesn't allow the node to determine if it's IPv4, IPv6, or FQDN. For ipv4 we used bytes. We should keep the `repeated ServiceEndpoint serviceEndpoint` list that we have from `NodeAddress`.

Reply:
Changed to use ServiceEndpoint type.